jccli/cli/group.py

```diff
@@ -14,29 +22,159 @@ def group(ctx):
-    # FIXME: Ideally, this would output JSON info of the deleted group (similar to delete-user), but at the moment it's
-    #  unclear exactly how to do that, given the API wrappers that we have.
+    logger = ctx.obj.get('logger')
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    group = api2.get_group(group_name=name, group_type=type)
+    if group is None:
+        logger.error(f"no group found of type '{type}', {name}")
+        sys.exit(1)
+    api2.delete_group(group['id'], type)
+    logger.info(f"successfully deleted group {name}")
+
+
+@group.command('add-user')
+@click.option('--name', "-n", required=True, type=str, help='name of the group')
+@click.option('--username', '-u', required=True, type=str, help='username of user to be added')
+@click.pass_context
+def add_user(ctx, name, username):
+    """
+    Add user to group (group type is assumed to be 'user_group')
+    """
+    logger = ctx.obj.get('logger')
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+
+    api1 = JumpcloudApiV1(ctx.obj.get('key'))
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    try:
+        user = api1.get_user(username)
+    except ApiException:
+        logger.info(f"no user found named {username}")
+        sys.exit(1)
+    group = api2.get_group(group_name=name, group_type=GroupType.USER_GROUP)
+    if group is None:
+        logger.error(f"no group of type '{type}' and name '{name}' found")
+        sys.exit(1)
+    try:
+        api2.bind_user_to_group(user['id'], group['id'])
+    except ApiException:
+        logger.error(f"API error (confirm that user {username} has not already been added to group {name}")
+        sys.exit(1)
+    logger.info(f"Successfully added user '{username}' to group '{name}'")
+
+
+@group.command('list-users')
+@click.option('--name', '-n', required=True, help='name of the group')
+@click.pass_context
+def list_users(ctx, name):
+    """
+    List users in a group (group type is assumed to be 'user_group')
+    """
+    api1 = JumpcloudApiV1(ctx.obj.get('key'))
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    logger = ctx.obj.get('logger')
+    group = api2.get_group(group_name=name, group_type=GroupType.USER_GROUP)
+    if group:
+        group_id = group['id']
+    else:
+        logger.error(f"no user group found with name {name}")
+        sys.exit(1)
+    user_ids = api2.list_group_users(group_id=group_id)
+    users = api1.retrieve_users(user_ids=user_ids)
+    serialized_response = json.dumps(users, indent=2)
+    logger.info(f"{serialized_response}")
+
+
+@group.command('remove-user')
+@click.option('--name', "-n", required=True, type=str, help='name of the group')
+@click.option('--username', '-u', required=True, type=str, help='username of user to be added')
+@click.pass_context
+def remove_user(ctx, name, username):
```

Comment:
Was `jsmith` indeed a member of the `staff` before you executed this? Otherwise, this looks like the desired behavior to me. Maybe the wording of the error message could be clearer?

Reply:
> Was jsmith indeed a member of the staff before you executed this? 

I don't think so.  jsmith user was created but was never placed into the staff group.

> Otherwise, this looks like the desired behavior to me. Maybe the wording of the error message could be clearer?

It wasn't clear to me what it meant initially but I think i understand now.  Yes,  lets clear this up with a some message like `error: User 'jsmith' is not in 'staff' group` or even the same message as the previous command `error: user 'jsmith' not found`?

Reply:
I'm trying to honestly relay the information that we're getting from the API to the user. When we make a call to remove a user from a group they're not in, we get an ApiException. I don't want to confidently tell the client that the user is not a member of the group. I imagine all kinds of issues could raise an ApiException.

The user not being a member of the group is the most likely cause of an ApiException, so I include that in a hint in parentheses, but I don't want to confidently report something that we don't know is true.

Reply:
I suppose we could parse the content of that ApiException, and infer the problem from there. This is what the content of a typical `ApiException` looks like when you try to remove a user from a group they're not in:

```
Exception when calling GraphApi->graph_user_group_members_post: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Server': 'nginx', 'Date': 'Fri, 04 Sep 2020 18:01:53 GMT', 'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '23', 'ETag': 'W/"17-SuRA/yvUWUo8rK6x7dKURLeBo+0"', 'X-Response-Time': '95.311ms'})
HTTP response body: {"message":"Not Found"}
```

My only hesitation is that I am just building on an observed result; as far as I can tell, the exact error messages (e.g. `"Not Found"`) you get for removing a user from a group they're not in (or adding a user to a group they're already in) [isn't specified](https://docs.jumpcloud.com/2.0/user-group-members-and-membership/manage-the-members-of-a-user-group), and therefore I'm hesitant to rely on them not changing the error message in the future.

Reply:
leave as-is for now.  maybe you can create an GH issue for this?  we can discuss it there and you can fix it later.